src/util/warn.js

```diff
@@ -16,10 +16,9 @@ export function isError (err: any): boolean {
   return Object.prototype.toString.call(err).indexOf('Error') > -1
 }
 
-export function isExtendedError (constructor: Function, err: any): boolean {
-  return (
-    err instanceof constructor ||
-    // _name is to support IE9 too
-    (err && (err.name === constructor.name || err._name === constructor._name))
-  )
+export function isRouterError (err: any, errorType: ?string): boolean {
+  const _isRouterError = isError(err) && err._isRouter
+
+  if (!errorType) return _isRouterError
+  return _isRouterError && err.type === errorType
```

Comment:
I think we can make this simpler:

```suggestion
  return isError(err) && err._isRouter && (errorType == null || err.type === errorType)
```